### ISSUE TITLE
KafkaJSProtocolError: The request attempted to perform an operation on an invalid topic

### ISSUE THREAD

**Issue by dennisLiu1991** [tulios/kafkajs#665]
Hi, we just found a weird problem about " The request attempted to perform an operation on an invalid topic".
we use dynamic topic like this:
```
function produce(type, content, response) {

  const topic = TOPIC_PREFIX + content.project + '-' + content.platform + '-' + type;
  let message = JSON.stringify(content);
  
  kafka.produce(topic, message).then(() => {
    output(response, 200, 'OK');
  }).catch(err => {
    output(response, 252, 'server error.');
  });
}
```
and when the value of _type_  contains special character, such as _type =  abc)(*&^%_ , it will throw an error like " KafkaJSProtocolError: The request attempted to perform an operation on an invalid topic".

that is all OK.

but after that , all produce operations , include the normal topic , can not work. the all throw a invalid topic error.

if we restart the server, the problem will fixed. but  if the wrong topic appear agian, and the problem will come back.

plz help us , thanks!

**Comment by Nevon** [comment]
Thanks for laying out the steps so clearly. I don't have any theory.off the top of my head that would explain why that would happen, as each produce is quite isolated from others. What would be really helpful would be a test that demonstrates the behavior.

**Comment by tulios** [comment]
It's poisoning the target group @Nevon; once the bad topic is there, all producers will have problems to refresh metadata and therefore throw the protocol error.

**Comment by dennisLiu1991** [comment]
thanks for replying , would you please tell me  how to solve this problem except restart?

**Comment by Nevon** [comment]
I don't believe there's anything you can do. Internally we need to remove the topic from the list of target topics in the cluster when we encounter this error. However in the meantime, this isn't something that you should need to recover from regularly - just make sure that you're not trying to produce to invalid topics.

**Comment by dennisLiu1991** [comment]
Thanks for your time, It is positive to see this is fixed so soon . I will try in my project and feed back  if this is any other problem : )

**Comment by Nevon** [comment]
This is fixed now in the pre-release version `1.13.0-beta.10`.
